### DISCUSSION TITLE
Applying different kernels on different sides of a moving interface

### DISCUSSION THREAD

**Discussion by WayneX-Wong** [idaholab/moose#24133] [Q&A General] [answered]
Hi all,

I'm trying to simulate kind of phase transition problem where interface moves based on calculated variable value, while governing PDEs of the subdomains on the two sides are different.

I found [XFEM Module](https://mooseframework.inl.gov/modules/xfem/) quite suitable for this kind of simulation about moving interface, but I wonder whether it support different kernels on different Cut-subdomains. I didn't find any similar tests or examples yet.

Or is there any alternative methods?

Thank you very much!

**Comment by jiangwen84** [comment, marked as answer]
For XFEM, currently we cannot restrict kernels to be calculated only on a specific subdomain.  One possible solution is to let kernels/physics to be solved on the whole domain, and then we just discard/ignore the solution that is relevant to that subdomain. I have not tried this myself yet, and you might encounter some issues. 

Alternatively, we would suggest you consider Phase Field approach for phase transition problem. It can eliminate a lot of troubles associated with moving the interface.

**Reply by hugary1995** [reply]
Did we ever merge the XFEM-based subdomain modifier?

**Reply by jiangwen84** [reply]
> Did we ever merge the XFEM-based subdomain modifier?

That is a good point. Maybe Yes. If so, we might be able to do it to some extend. It is not fully tested, and I am afraid there might be many issues here and there.
